qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -132,6 +132,26 @@ def __init__(
         self.tol = tol
         self.callback = callback
 
+    @property
+    def settings(self) -> Dict[str, Any]:
+        if self.callback is not None:
+            raise ValueError("Cannot serialize GradientDescent with callback.")
```

Comment:
We can keep `settings` generic and still return `callback` in the dictionary. The JSON encoder will just fail when it encounters a callable (although the error message won't be as obvious).

Reply:
yes, done 👍🏻